app_flutter/lib/service/fake_cocoon.dart

```diff
@@ -20,7 +22,7 @@ class FakeCocoonService implements CocoonService {
 
   @override
   Future<bool> fetchTreeBuildStatus() async {
-    return true;
+    return Random().nextBool();
```

Comment:
Instead  of making this random  -  and therefore untestable, maybe make the value configurable for the fake?

Reply:
I added a way  to inject it, so tests can inject their own "random" generator.